### ISSUE TITLE
Behavior of `*` could be improved

### ISSUE THREAD

**Issue by quadruple-output** [dexterleng/homerow#10]
I propose to either change how `*` works, or provide a configuration option as follows:

- Typing `*` would not be required at all if an empty search field would highlight all targets immediately (my preferred solution).
- As a config option, the character `*` could be defaulted (pre-filled) in the search field, and pre-selected, such that users don't have to explicitly delete the `*` letter before typing the actual search term.

## Motivation

I am currently using the following tools which all offer similar, but slightly different functionality:
- `homerow`, because I hope it will become the future single replacement for all the apps below;
- `vimac`, because it works well for cases where `homerow` does not (yet) offer a solution;
- `vimium` plugin for Firefox, because it is the only solution that works within Firefox;
- `vimari` plugin for Safari, but it seems that `homerow` can already replace it for my purposes (which is great!!).

My current problem with `homerow` is that I cannot (yet) predict very well, which targets it offers. Therefore, I find myself mostly using `*` first, to see whether the UI element I want to address even has a target. Next, I have to *delete* the `*` character again, before I can type the search term. This feels quite unnecessary. It would be easier if the available targets would be shown from the start, and I would not need to manually delete the `*` before I can search.

(I think) I understand the motivation why the current implementation does not highlight all targets by default. For very experienced users, this reduces the noise on the screen. However, for users who are just learning how to use the app efficiently, any visual support would be beneficial. This may also be the reason why not everyone likes the change from the old way of assigning arbitrary (and changing) key combinations to the targets, to the new way of using texts that are already provided by the targets themselves. The old way is easy to understand, while the new way may (sometimes) be faster for experienced users.

(I really hope that many people buy a `homerow` license, so we can expect the app to come to maturity.)

**Comment by dexterleng** [comment]
@quadruple-output I totally agree! I occasionally forget to clear the non-wildcard input after a wildcard and vice versa.

I have an unreleased preview available that addresses both (1) autoclearing and (2) showing targets by default, would love to hear your feedback!

https://github.com/dexterleng/homerow/issues/6#issuecomment-1319994219

**Comment by dexterleng** [comment]
🚀 Shipped in Alpha 9. Closing this thread. Lemme know if you have any feedback.

**Comment by quadruple-output** [comment]
Sorry for not getting back to you earlier. 😞 

I updated to Alpha 9. 

**`*` character:** Very Good! I don't know why you chose `;` as a second alternative, but why not. As far as I see, using `*` is not required, any more – except for clearing the search term for a fresh start. I am not sure if this rare use case justifies the special semantics of `*` in the first place. If I need to clear my search term, I can also hit `Escape` and restart. For the sake of simplicity, my personal preference would be not to have a special semantics for `*` (or `;`). (But I do not have a strong opinion, here.)

**Search Mode vs. "Only Labels (classic)":** I must say that I am very much used to the classic mode from Vimac, so I am tempted to use this mode in Alpha 9, as well (though I will continue to use search mode until I found a decision for me whether it has benefits). I have two small remarks on the classic mode: 

a) I personally favor the label position of Vimac (centered) over Homerow (right-aligned), but both versions have the issue that, for larger targets, the labels are sometimes hard to spot. Alpha 8 highlighted all targets with an overlay of the size of the target. I liked this very much. Alpha 9 does not do this any more, but relies on labels only. I am not sure what the best approach would be, but at least for larger targets, the overlay can be very helpful.

b) When I chose "Labels Only (classic)" in the preferences, I still have to confirm each key combination with `Enter`. This is one key stroke more than necessary.

**Comment by quadruple-output** [comment]
I just changed my mind about the `*` character being not required... I just discovered this work flow for me:
In Safari, on a web page with many targets, I want to click a link without a label. This link is close to another target which does have a label. So, I enter the first characters of the nearby label to focus it. Now, I need only a few `tabs` to focus on the link I actually want to click. So, I press `;` (which actually is easier for me to hit than `*`, btw. 😸) to remove the filter on the available targets, and press `tab` a few times.

There is only one problem: pressing `;` or `*` changes the focus back to the first target. I would rather not have the focus changed, so I can continue tabbing from where I focussed on.
